generator.rb

```diff
@@ -0,0 +1,93 @@
+# Portmanteau Generator
+
+def is_vowel?(letter)
+  if ["a", "e", "i", "o", "u"].include?(letter)
+    return true
+  else
+    return false
+  end
+end
+
+def run_generator
+  # prompt user for first word
+  puts "What is the first word to combine?"
+  first_word = gets.chomp
+
+  # input validation for first word
+  while first_word.length < 2
+    puts "We need words with two letters or more!"
+    puts "Please re-enter the first word: "
+    first_word = gets.chomp
+  end
+
+  # prompt user for second word
+  puts "What is the second word to combine?"
+  second_word = gets.chomp
+
+  # input validation for second word
+  while second_word.length < 2
+    puts "We need words with two letters or more!"
+    puts "Please re-enter the first word: "
+    second_word = gets.chomp
```

Comment:
Your code to get the second word is almost exactly the same as the code to get the first word. Could you DRY this up using another method?

Reply:
Notes for myself:

def input_validation(word)
  while word.length < 2
    puts "We need words with two letters or more!"
    puts "Please re-enter the first word:  "
    word = gets.chomp
  end
end